flatland_server/include/flatland_server/model.h

```diff
@@ -52,34 +52,38 @@
 #include <flatland_server/joint.h>
 #include <flatland_server/model_body.h>
 #include <flatland_server/yaml_reader.h>
+#include <flatland_server/message_system/message_server.h>
 #include <yaml-cpp/yaml.h>
 #include <boost/filesystem.hpp>
 
 namespace flatland_server {
 
 class ModelBody;
 class Joint;
+class World;
 
 /**
  * This class defines a Model. It can be used to repsent any object in the
  * environment such robots, chairs, deskes etc.
  */
 class Model : public Entity {
+ private:
+  flatland_server::World *world;
```

Comment:
Is there a possibility that the world is destroyed before the model?

Reply:
World gets destroyed and calls destructors for the list of models it contains.

Reply:
cool!